interface/cpp.cc

```diff
@@ -371,6 +372,31 @@ void cpp_generator::print_methods_decl(ostream &os, const isl_class &clazz)
 		print_method_group_decl(os, clazz, it->first, it->second);
 }
 
+/* Print declarations for custom members of a class "clazz" to "os", based on
+ * the class name.
+ */
+void cpp_generator::print_custom_public_decl(ostream &os,
+	const isl_class &clazz)
+{
+	string cppname = type2cpp(clazz);
+
+	if ("id" == cppname) {
+		const char *declarations =
+			"  inline id(isl::ctx ctx, const std::string &name);\n"
+			"  inline id(isl::ctx ctx, const std::string &name,\n"
+			"            void *usr,\n"
+			"            void (*deleter)(void *) = nullptr);\n"
+			"  inline id(isl::ctx ctx, void *usr,\n"
+			"            void (*deleter)(void *) = nullptr);\n"
```

Comment:
The functions you choose look good. I am slightly afraid that pretty printing them like this won't fly. At the very least, the first one could be declared as C function and normally exposed. This would make sure that isl ids are also available in python.

Not sure about the other two. Do they make any sense in python? I mean, do we want to pass references to other python objects as user point? (A deleter certainly does not make sense)